(#49) Add migration page (#50)
chocolatey-community/chocolatey-au: Migrating-AU-to-Chocolatey-AU.md

# Migrating From `AU` To `Chocolatey-AU`

With the [`1.0.0` release](https://github.com/chocolatey-community/chocolatey-au/releases/tag/1.0.0) of `Chocolatey-AU`, there is a need to migrate to the new package name.

> How you migrate from `AU` to `Chocolatey-AU` will depend on how you used `AU` in your repository. Below are some scenarios, if you encounter a scenario not yet covered, please reach out in the [Chocolatey Community](https://ch0.co/community).

Reviewer: ```suggestion
How you migrate from `AU` to `Chocolatey-AU` will depend on how you used `AU` in your repository. Below are some scenarios, but if you encounter a scenario not yet covered, please reach out in the [Chocolatey Community](https://ch0.co/community).

```

> ## Your Repository Currently Clones The Original `AU` Repository

Reviewer: ```suggestion
## Your Repository Currently Clones the Original `AU` Repository

```

> If you are cloning the [original `AU` repository](https://github.com/majkinetor/au/) and building the module from there, you will want to update to clone the Chocolatey Community `Chocolatey-AU` repository: `https://github.com/chocolatey-community/chocolatey-au.git` instead. Once that has been done, you will want to [update your update scripts](#updating-your-update-scripts) to use the `Chocolatey-AU` PowerShell Module instead of `AU`.

Reviewer: ```suggestion
If you are cloning the [original `AU` repository](https://github.com/majkinetor/au/) and building the module from there, you will want to update to clone the Chocolatey Community `Chocolatey-AU` repository: `https://github.com/chocolatey-community/chocolatey-au.git` instead. Once that has been done, you will want to [amend your update scripts](#updating-your-update-scripts) to use the `Chocolatey-AU` PowerShell Module instead of `AU`.

```

> ## Your Repository Installs AU From The Chocolatey Community Repository

Reviewer: ```suggestion
## Your Repository Installs AU From the Chocolatey Community Repository

```

> If you are installing `AU` from the [Chocolatey Community Repository](https://community.chocolatey.org/), you will want to update it to `choco install chocolatey-au --confirm` instead of `choco install au --confirm`. Once that has been done, you will want to [update your update scripts](#updating-your-update-scripts) to use the `Chocolatey-AU` PowerShell Module instead of `AU`.

Reviewer: ```suggestion
If you are installing `AU` from the [Chocolatey Community Repository](https://community.chocolatey.org/), you will want to update it to `choco install chocolatey-au --confirm` instead of `choco install au -y`. Once that has been done, you will want to [update your update scripts](#updating-your-update-scripts) to use the `Chocolatey-AU` PowerShell Module instead of `AU`.

```

Reviewer: When writing code, refer to the [Writing Documentation](https://design.chocolatey.org/content-and-marketing/writing-documentation#code-style) guide and the [Scripting and Best Practices Style Guide](https://docs.chocolatey.org/en-us/choco/commands/#scripting-integration-best-practices-style-guide). 

Specifically here, we don't use `--confirm`, we use `-y`.

> ## Your Repository Installs AU From The Powershell Gallery

Reviewer: ```suggestion
## Your Repository Installs AU From the Powershell Gallery

```

> If you are installing `AU` from the [PowerShell Gallery](https://powershellgallery.com), you will want to update it to `Install-Module Chocolatey-AU` instead of `Install-Module AU`. Once that has been done, you will want to [update your update scripts](#updating-your-update-scripts) to use the `Chocolatey-AU` PowerShell Module instead of `AU`.

Reviewer: ```suggestion
If you are installing `AU` from the [PowerShell Gallery](https://powershellgallery.com), you will want to amend your code to `Install-Module Chocolatey-AU` from `Install-Module AU`. Once that has been done, you will want to [amend your update scripts](#updating-your-update-scripts) to use the `Chocolatey-AU` PowerShell Module instead of `AU`.

```

> ## Updating Your Update Scripts

Reviewer: ```suggestion
## Amending Your Update Scripts

```

> Once you have updated your repository to use the new module name, you will need to update your `update.ps1` scripts to use the new module name. This is as simple as replacing `AU` with `Chocolatey-AU` for any call to `Import-Module`, or any `#requires -Modules` in your scripts.

Reviewer: ```suggestion
Once  your repository uses the new module name, you will need to amend your `update.ps1` scripts to use the new module name. This is as simple as replacing `AU` with `Chocolatey-AU` for the `Import-Module` calls, or any `#requires -Modules` in your scripts.

```

# Migrating From `AU` To `Chocolatey-AU`

With the [`1.0.0` release](https://github.com/chocolatey-community/chocolatey-au/releases/tag/1.0.0) of `Chocolatey-AU`, there is a need to migrate to the new package name.

How you migrate from `AU` to `Chocolatey-AU` will depend on how you used `AU` in your repository. Below are some scenarios, but if you encounter a scenario not yet covered, please reach out in the [Chocolatey Community](https://ch0.co/community).

## Your Repository Currently Clones the Original `AU` Repository

If you are cloning the [original `AU` repository](https://github.com/majkinetor/au/) and building the module from there, you will want to update to clone the Chocolatey Community `Chocolatey-AU` repository: `https://github.com/chocolatey-community/chocolatey-au.git` instead. Once that has been done, you will want to [amend your update scripts](#amending-your-update-scripts) to use the `Chocolatey-AU` PowerShell Module instead of `AU`.

> 

Reviewer 2: Small nitpick. There should only be one empty line (not too important though).

Author: I've removed the extra line.

# Migrating From AU To Chocolatey AU

> With the [`1.0.0` release] of Chocolatey AU, there is a need to migrate to the new package name.

Reviewer 2: If we will be going forward with reference linking (based on Paul's comment, we won't though), let's use it according to specs.
The specs define them as needing to have two square brackets, so in this case it would be

``[`1.0.0` release][]`` (the first one is the text being displayed, and the second being the name of the link which will use the text if it is empty).

How you migrate from AU to Chocolatey AU will depend on how you used AU in your repository. Below are some scenarios, if you encounter a scenario not yet covered, please reach out in the [Chocolatey Community].

> ## Your Repository Currently Clones The [Original AU Repository]

Reviewer 2: Let's not start making use of links in the header, while it will work in most cases, there can be times it won't work properly.

Author: That's fair. I've removed it from the headings.

## Your Repository Installs AU From The [Chocolatey Community Repository]

If you are installing `AU` from the [Chocolatey Community Repository], you will want to update it to `choco install chocolatey-au --confirm` instead of `choco install au --confirm`. Once that has been done, you will want to [update your update scripts](#updating-your-update-scripts) to use the `Chocolatey-AU` PowerShell Module instead of `AU`.

## Your Repository Installs AU From The [Powershell Gallery]

If you are installing `AU` from the [PowerShell Gallery], you will want to update it to `Install-Module Chocolatey-AU` instead of `Install-Module AU`. Once that has been done, you will want to [update your update scripts](#updating-your-update-scripts) to use the `Chocolatey-AU` PowerShell Module instead of `AU`.

## Updating Your Update Scripts

> Once you have updated your repository to use the new module name, you will need to update your `update.ps1` scripts to use the new module name. This is as simple as replacing `AU` with `Chocolatey-AU` for any call to `Import-Module`, or any `#requires -Modules` in you `update.ps1` scripts.

Reviewer 3: Also worth mentioning that if there is an `import-module au` in the `update_all.ps1`, it should be updated as well.

# Migrating From `AU` To `Chocolatey-AU`

With the [`1.0.0` release](https://github.com/chocolatey-community/chocolatey-au/releases/tag/1.0.0) of `Chocolatey-AU`, there is a need to migrate to the new package name.

How you migrate from `AU` to `Chocolatey-AU` will depend on how you used `AU` in your repository. Below are some scenarios, but if you encounter a scenario not yet covered, please reach out in the [Chocolatey Community](https://ch0.co/community).

> ## Your Repository Currently Clones the Original `AU` Repository

Author: ```suggestion
## Your Repository Currently Forks the Original `AU` Repository
```

Reviewer 2: Most likely not the appropriate change. Clones would still be correct here.

Forking the repository would be creating a copy of the repository into a new repository, but most likely they are just taking the original repository and cloning it to a local instance (technically, a fork is also a clone, but a clone is not necessarily a fork).

Author: Good catch. This was based off of some discussions in Discord. I'll add some clarifications about the cloning part to make it clearer.
